### PR TITLE
Enforce 'strict' flag for Typescript 

### LINKED ISSUE #116: [Feature Request] - Enforce 'strict' flag for Typescript
### Pre-requisites:
- [x] I looked through both [open and closed issues](../issues?utf8=✓&q=is%3Aissue) and did not find another request for the same or similar feature.

## Describe the solution you'd like

We should enable `strict` mode in `tsconfig.json` and correct all the errors that come up.

### PR DESCRIPTION
**Pull request by smitshah2531**
## Pre-Requisites
- [x] Yes, I updated [Authors.md](../blob/main/Authors.md) **OR** this is not my first contribution
- [x] Yes, I included and/or modified tests to cover relevent code **OR** my change is non-technical
- [x] Yes, I wrote this code entirely myself **OR** I properly attributed these changes in [Third Party Notices](../blob/main/THIRD-PARTY-NOTICES.txt)

## Description of Changes

Updated strict to true in tsconfig.json file

@smitshah2531 worked with Joseph on this PR

## Related Issues

This change closes #116 which is asking to enforce a strict flag for Typescript

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -30,6 +30,9 @@ This app is still in the early stages of development and only some functionality
 ---
 
 # Local Development
+<<<<<<< HEAD
```
Comment (SpencerKaiser) [inline README.md:33]: This was a bad merge conflict resolution. Have you ever had to resolve a merge conflict before?
```suggestion
```

File: README.md [hunk 2]
```diff
@@ -30,6 +30,9 @@ This app is still in the early stages of development and only some functionality
...
+Hangar uses Postgres 11, so you'll need to set it up on your machine. During the installation process, you will be prompted to create a password. You need to create PGUSER and PGPASSWORD in your `.env` file where PGUSER will be `postgres` and PGPASSWORD will be the password you enter upon installation. Create a database (we suggest `hangar`, if you chose something else or are running your server on a different port, make sure to create a `DATABASE_URL` value in `.env` with your override URL) in the PostgreSQL 11 server, ensure it is not a seperate server, and then run the app. When the app is deployed to a cloud environment, the `DATABASE_URL` `.env` var will be used (and is automatically set in Heroku when an associated service is connected to your app). 
+=======
```
Comment (SpencerKaiser) [inline README.md:35]:
```suggestion
```

File: README.md [hunk 3]
```diff
@@ -50,6 +53,7 @@ During the installation process (if you follow the steps on [postgresql.org](htt
 
 #### Viewing/Editing the DB
 If you'd like a visual way of viewing or editing your local database, try using [TablePlus](https://tableplus.com).
+>>>>>>> 4a6ac4c3093279a84a33747efe3535c2bc9be127
```
Comment (SpencerKaiser) [inline README.md:56]:
```suggestion
```

**Review by SpencerKaiser — CHANGES_REQUESTED** [review]
In addition to my requested changes, it also looks like the build is failing...

File: README.md (hunk 1 above)
Comment (smitshah2531) [reply README.md:33]: Yes. I had to resolve a merge conflict before. I will make the suggested changes.

**Comment by smitshah2531** [comment]
> In addition to my requested changes, it also looks like the build is failing...

Yes. I am trying to figure out why build is failing.

**Comment by SpencerKaiser** [comment]
@smitshah2531 did you check the [GitHub Action for PR-Build](https://github.com/AmericanAirlines/Hangar/pull/150/checks?check_run_id=1125960104)? That's what's failing:
image


Also next time, please read the full issue description template. You delete the section for related issues, which is very important:
```## Related Issues
<!-- Include a list and brief description of any tracked issues -->
<!-- e.g., "Fixes #123 - A bug that crashes the app" -->
```

Please adjust the description to include the above. Also please note that the `<!-- -->` syntax isn't how you're supposed to format the text, those are just hidden comments so that text isn't shown when you actually create the PR 😉

**Comment by smitshah2531** [comment]
> @smitshah2531 did you check the [GitHub Action for PR-Build](https://github.com/AmericanAirlines/Hangar/pull/150/checks?check_run_id=1125960104)? That's what's failing:
> image
> 
> Also next time, please read the full issue description template. You deleted the section for related issues, which is very important:
> 
> ```
> ## Related Issues
> 
> 
> ```
> 
> Please adjust the description to include the above. Also please note that the `<!-- -->` syntax isn't how you're supposed to format the text, those are just hidden comments so that text isn't shown when you actually create the PR 😉

Yes. I see why it is failing. Working on it right now. Also, I updated the description above. It makes a lot of sense now why we need to include "Related Issue". Sorry for the confusion :)

**Comment by SpencerKaiser** [comment]
@smitshah2531 I goofed up... the issue with this is that we updated the _wrong tsconfig file_. You need to move that `strict` flag to `api/tsconfig.json` then fix all the errors...

**Comment by smitshah2531** [comment]
> @smitshah2531 I goofed up... the issue with this is that we updated the _wrong tsconfig file_. You need to move that `strict` flag to `api/tsconfig.json` then fix all the errors...

Ah, I see. I will do that and see if that works. Thanks :)

**Comment by smitshah2531** [comment]
@SpencerKaiser just to make sure you mean `src/tsconfig.json` correct? Because I don't see any config.json file such as `api/tsconfig.json`

**Comment by smitshah2531** [comment]
I fixed the above error for the README.md file by pulling the latest changes from the main so I am resolving it.

**Comment by SpencerKaiser** [comment]
@smitshah2531 yes, `src/tsconfig.json`. Thanks for clarifying!

**Comment by smitshah2531** [comment]
@SpencerKaiser I am still getting the same errors. There are lots of errors in multiple files, I solved some of them but still I am not sure I am going on right track or not. If you don't mind then I would like to sync together and work from there onwards :)

**Comment by SpencerKaiser** [comment]
Do your best to fix the errors, I can provide feedback via a PR review. If something is undefined, you can always throw an error if you think it's the most appropriate way to handle the issue:
``` typescript
  if (!judge) {
    logger.crit("Somehow we don't have a judge for the vote being cast");
    throw new Error('Judge not found');
  }
```

**Comment by SpencerKaiser** [comment]
@johnkahn and I will work on this at a later date

**Pull request closed without merging**